adapter_darwin.go

```diff
@@ -156,6 +156,19 @@ func makeScanResult(prph cbgo.Peripheral, advFields cbgo.AdvFields, rssi int) Sc
 		})
 	}
 
+	var serviceData []ServiceDataElement
+	for _, svcData := range advFields.ServiceData {
+		cbgoUUID := svcData.UUID
+		uuid, err := ParseUUID(cbgoUUID.String())
+		if err != nil || uuid.String() != cbgoUUID.String() {
```

Comment:
Why the extra check whether the UUID matches? (if it doesn't, it means the UUID either is formatted differently for example due to capitalization, or there's a bug in `ParseUUID` which isn't the responsibility of the calling code).

```suggestion
		if err != nil {
```

(Also, there _should_ never be an error here but as it's an external API it doesn't hurt to check for the error response anyway).

Reply:
actually that piece came with the other PR in the beginning. but sure that check is redundant.